kernel/debugmalloc.c

```diff
@@ -173,6 +177,12 @@ MYST_INLINE void* _get_block_address(void* ptr)
     return (uint8_t*)ptr - sizeof(header_t) - padding_size;
 }
 
+MYST_INLINE void* _get_block_address_v2(void* ptr)
```

Comment:
Do we still need `_get_block_address` v1?

Reply:
We could eliminate v1, but I left it as a cross check in case the two functions fail to produce the same quantity. The v2 version is just intended as a sanity check (so if v1 != v2 then we return EINVAL).